### PR TITLE
PassTicket REST API

### PR DESCRIPTION
**Pull request by lchudinov**
# PassTicket REST API

## Proposed changes
This PR adds an endpoint for PassTicket generation.

This PR addresses Issue: https://github.com/zowe/zlux/issues/683

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## PR Checklist
- [x] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [x] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Relevant update to CHANGELOG.md
- [x] My changes generate no new warnings

## Testing
New endpoint is `/passticket`.  
### Request
`POST /passticket`
Request body:
```json
{
  "applicationName": "ZWEAPP"
}
```
### Response
`HTTP 200 OK` 
Response body:
```json
{
  "userId": "USERID",
  "applicationName": "ZWEAPP",
  "ticket": "09X0CCGJ"
}
```
### Error response:
`500 Interval Server Error`
Response body:
```json
{
  "error": "Failed to generate PassTicket"
}
```

### CODE REVIEW

File: c/passTicketService.c [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+#include "charsets.h"
+#include "httpserver.h"
+#include "json.h"
+#include "zssLogging.h"
+#include "passTicketService.h"
+#include "serviceUtils.h"
+
+#define TICKET_LEN 8
+#define APP_NAME_KEY "applicationName"
+#define APP_NAME_KEY_LEN 8
```
Comment (DivergentEuropeans) [inline c/passTicketService.c:30]: Is there a reason we are limiting app name to 8? I don't see any such restriction in:

https://docs.zowe.org/stable/extend/extend-apiml/api-mediation-passtickets/

Comment (lchudinov) [reply c/passTicketService.c:30]: Yes, application name is only 1-8 byte long: https://www.ibm.com/docs/en/zos/2.4.0?topic=3-generate-passticket1-4

Comment (DivergentEuropeans) [reply c/passTicketService.c:30]: Nevermind, just found elsewhere "The APPLID is the name of the application (up to 8 characters) that is used by security products to differentiate certain security operations (like PassTickets) between applications."

**Review by DivergentEuropeans — COMMENTED** [review]
Functionally, this works for me. I also get an error from RACF when I try to generate one which is what I would expect. Other errors like APPID > 8, I also see. Can't speak to whether this works as intended, someone from API ML team needs to verify that

Code seems to look good

**Review by 1000TurquoisePogs — APPROVED** [review]
This looks good to me, but I have not tested it yet.

**Comment by achmelo** [comment]
I have verified this functionality with the integration to API ML and everything worked as described.

**Pull request merged**